Week3/homework/js-exercises/Project-2-JS1/creditCardValid.js

```diff
@@ -0,0 +1,26 @@
+'use strict'
+
+function cardValidator(cardNum) {
+    let result = "";
+    const digits = cardNum.split("");
+    const lastDigit = digits[digits.length - 1];
+    const sumOfDigits = digits.reduce((total, item) => {
+        return Number(total) + Number(item)
+    });
+    if (cardNum.toString().length !== 16 || Number.isInteger(Number(cardNum)) === false) {
+        result = `${cardNum} invalid characters or length`;
+    } else if (lastDigit % 2 === 1) {
+        result = `${cardNum} odd final number`;
+    } else if (digits.filter(item => {
+            return item === lastDigit
+        }).length == digits.length) {
```

Comment:
👏 for the logic. However, to find out if we have atleast two unique numbers, I would like to introduce to you a new addition in JavaScript called `Sets`. A **set** can only contain unique entries. Thus, instead of using the `filter` method here, what you could also do is:

```
const uniqueLength = new Set(Array.from(`${cardNum}`).size; // uniqueLength will be 1 if there are no unique characters and will be more than 1 if there are atleast two. You can use this instead of a filter method in such a case 🙂 
```